src/sema/mod.rs

```diff
@@ -386,6 +408,122 @@ fn resolve_pragma(
     }
 }
 
+/// Walk through the parse tree and collect all the annotations and doccomments for
+/// each item, also inside contracts.
+fn collect_annotations_doccomments<'a>(
```

Comment:
Nit: Ideally this would be 2 distinct functions

Reply:
I can split the function, but this would mean two iterations over the parse tree.

Reply:
Fair enough.

Since those two things do not seem to relate or depend on each other, I guess a more ideal solution would be to walk the tree twice, but in parallel. But it is not the time to open this whole high level design discussion again :sweat_smile: